### DISCUSSION TITLE
Remotely start the Slint application on linux using ssh

### DISCUSSION THREAD

**Discussion by kevin406972552** [slint-ui/slint#3305] [Q&A]
**Development PC**
```
kevin@ubuntu:~$ ssh root@192.168.137.134
The authenticity of host '192.168.137.134 (192.168.137.134)' can't be established.
ECDSA key fingerprint is SHA256:XcYCPoVBEWw3jzK0Fpdl0yhqr4BqbmfLt8nwkuKD1Nw.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '192.168.137.134' (ECDSA) to the list of known hosts.
root@192.168.137.134's password: 
  ___  ____  _   _____             ____  
 / _ \|  _ \(_) |__  /___ _ __ ___|___ \ 
| | | | |_) | |   / // _ \ '__/ _ \ __) |
| |_| |  __/| |  / /|  __/ | | (_) / __/ 
 \___/|_|   |_| /____\___|_|  \___/_____|
                                         
Welcome to Orange Pi 3.0.6 Jammy with Linux 5.16.17-sun50iw9

System load:   25%           	Up time:       24 min	Local users:   2            	
Memory usage:  55% of 984M   	Zram usage:    28% of 492M   	IP:	       192.168.137.134
CPU temp:      56°C           	Usage of /:    23% of 15G    	

[ 31 security updates available, 87 updates total: apt upgrade ]
Last check: 2022-08-09 13:16

New to Orange Pi? Support: http://www.orangepi.org

root@orangepizero2:~# echo $DISPLAY

root@orangepizero2:~# export DISPLAY=:0.0
root@orangepizero2:~# cd /
root@orangepizero2:/# ./slint-2 
Authorization required, but no authorization protocol specified
thread 'main' panicked at 'Failed to initialize any backend! Wayland status: NoCompositorListening X11 status: XOpenDisplayFailed', /home/kevin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.28.6/src/platform_impl/linux/mod.rs:757:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
root@orangepizero2:/# 
```

**Embedded device**
![image](https://github.com/slint-ui/slint/assets/25196508/d75f5cfd-0f88-4cb8-975b-571e2fb326ec)

**Comment by hunger** [comment]
This looks like a generic X window issue to me:

Quick and dirty work-around: Run `xhost +` _before_ you do SSH (or in another terminal window). That *disables authorization entirely* and thus is an option only when your X server is not accessible from the network (most are not nowadays, but do not use this in production ever).

The secure solution involves messing with `xauth`, but I forgot how to do that exactly.

This above assumes the X server is running on the maschine you SSH into. If that is not the case: You need to configure your X server to listen for network connections (most do not anymore), and you can not use `DISPLAY=:0.0` as that is local to the client... You need something like `DISPLAY=SERVER_IP:0.0` or something IIRC. It's been a while since I used remote X forwarding.

Note: Whatever you do: Forwarding X applications over the network is pretty slow.

**Reply by tronical** [reply]
A possibly better solution would be to not log in as root (`ssh root@192.168.137.134`) but as user `orangepi` (`ssh orangepi@192.168.137.134`). Or is there a specific reason why you're logging in as root but the windowing session is running under the `orangepi` user?
